[MA-34] - RDS Audit table

balkrishnada opened the pull request:
https://jenkins.media.dev.usa.reachlocalservices.com/view/media-automation/job/media-automation-build/26/

jellyvora-reachlocal commented:
@balkrishnada Add the table to test init sql too

balkrishnada commented:
Let it get finalised will add

RyanGunapalaRL commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,19 @@
+    `keyword`                varchar(255)     NOT NULL,
+    `geo_keyword`            varchar(255)     NOT NULL,
+    `match_type`             varchar(14)      NOT NULL,
+    `status`                 enum ('DELETED','ALREADY_INACTIVE','ALREADY_QUEUED_FOR_INACTIVE','KEYWORD_MISSING_ON_CID') DEFAULT NULL,
```
You might wanna have some sort of catch all status like unknown with exception in a message column?

balkrishnada replied:
@RyanGunapalaRL that's a good point.

Will it be better to extract the `comments` column into a new table and link by `id`. Since its going to be empty for most of the rows.

RyanGunapalaRL replied:
no, let's keep it simple, we don't have a storage problem

balkrishnada commented:
**new updates:**
- Introduced new column `botcommand_id` so that we know what bot command was issued for this removal. An important piece of data.
- Renamed 'DELETED` to `BOTCOMMAND_CREATED`
- New ENUM `EXCEPTION`
- New column `message` to store the exception.

RyanGunapalaRL commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,23 @@
+    `keyword`                varchar(255)     NOT NULL,
+    `geo_keyword`            varchar(255)     NOT NULL,
+    `match_type`             varchar(14)      NOT NULL,
+    `status`                 enum ('BOTCOMMAND_CREATED','ALREADY_INACTIVE','ALREADY_QUEUED_FOR_INACTIVE',
```
consider these values:
```SUCCESS
KEYWORD_MISSING
KEYWORD_INACTIVE
ALREADY_REQUESTED_FOR_DELETION
FAILURE
```

balkrishnada replied:
done

RyanGunapalaRL commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,23 @@
+    `id`                     int(10) unsigned NOT NULL AUTO_INCREMENT,
+    `batch_job_execution_id` bigint(20)       NOT NULL,
+    `botcommand_id`          int(10)          NULL,
+    `campaign_id`            int(10)          NOT NULL,
```
i would also include a `mcid` column

balkrishnada replied:
done

RyanGunapalaRL commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,23 @@
+    `geo_keyword_id`         int(10)          NOT NULL,
+    `keyword`                varchar(255)     NOT NULL,
+    `geo_keyword`            varchar(255)     NOT NULL,
+    `match_type`             varchar(14)      NOT NULL,
```
is match_type id more important?

balkrishnada replied:
yes, match_type information will be helpful to know what it was when the keyword was deleted.

balkrishnada commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE `low_qs_keyword_auditlog`
```
done

balkrishnada commented:
New successful build: https://jenkins.media.dev.usa.reachlocalservices.com/view/media-automation/job/media-automation-build/37/
